ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/OperationsReOrgManager.java

```diff
@@ -0,0 +1,143 @@
+      AttestationManager attestationManager,
+      RecentChainData recentChainData) {
+    this.exitPool = exitPool;
+    this.proposerSlashingPool = proposerSlashingPool;
+    this.attesterSlashingPool = attesterSlashingPool;
+    this.attestationManager = attestationManager;
+    this.attestationPool = attestationPool;
+    this.recentChainData = recentChainData;
+  }
+
+  @Override
+  public void reorgOccurred(
+      Bytes32 bestBlockRoot, UInt64 bestSlot, Bytes32 oldBestBlockRoot, UInt64 commonAncestorSlot) {
+    NavigableMap<UInt64, Bytes32> notCanonicalBlockRoots =
+        recentChainData.getEveryRootOnChainTillSlot(commonAncestorSlot, oldBestBlockRoot);
+    notCanonicalBlockRoots.forEach(
+        (__, root) -> {
+          SafeFuture<Optional<BeaconBlock>> maybeBlockFuture =
+              recentChainData.retrieveBlockByRoot(root);
+          maybeBlockFuture
+              .thenAccept(
+                  maybeBlock ->
+                      maybeBlock.ifPresentOrElse(
+                          block -> {
+                            BeaconBlockBody blockBody = block.getBody();
+                            proposerSlashingPool.addAll(blockBody.getProposer_slashings());
+                            attesterSlashingPool.addAll(blockBody.getAttester_slashings());
+                            exitPool.addAll(blockBody.getVoluntary_exits());
+
+                            // Attestations need to get re-processed through AttestationManager
+                            // because we don't have access to the state with which they were
+                            // verified anymore and we need to make sure later on
+                            // that they're being included on the correct fork.
+                            blockBody
+                                .getAttestations()
+                                .forEach(
+                                    attestation -> {
+                                      attestationManager
+                                          .onAttestation(attestation)
+                                          .finish(
+                                              result ->
+                                                  result.ifInvalid(
+                                                      reason ->
+                                                          LOG.debug(
+                                                              "Rejected re-queued attestation from block: {} due to: {}",
+                                                              root,
+                                                              reason)),
+                                              err ->
+                                                  LOG.error(
+                                                      "Failed to process re-queued attestation from block: {} due to: {}",
+                                                      root,
+                                                      err));
```

Comment:
```suggestion
                                                  LOG.error(
                                                      "Failed to process re-queued attestation from block: {}",
                                                      root,
                                                      err));
```

Avoid losing the error message.

Reply:
Done.